File: src/gof/AbstractFactory/MazeGameMazeFactory.cpp
Maze* MazeGame::CreateMaze (MazeFactory& factory){
    Maze* aMaze = factory.MakeMaze();
    Room* r1 = factory.MakeRoom(1);
    Roam* r2 factory.MakeROOm(2);
    Door* aDoor factory.MakeDoor (r1, r2);

    aMaze->AddRoom(r1);
    aMaze->AddRoom(r2);

    r1->SetSide(North, factory.Makewall());
    r1->SetSide(East, aDoor);
    r1->SetSide(South, factory.Makewall());
    r1->SetSide(West, factory.Makewall());

    r2->SetSide(North, factory.Makewall());
    r2->SetSide(East, factory.Makewall());
    r2->SetSide(South, factory.Makewall());
    r2->SetSide(West, aDoor);
    
    return aMaze;
}

File: src/gof/AbstractFactory/code/MazeGameMazeFactory.java
public Maze createMaze (MazeFactory factory){
    Maze aMaze = factory.makeMaze();
    Room r1 = factory.makeRoom(1);
    Roam r2 factory.makeRoom(2);
    Door aDoor factory.makeDoor (r1, r2);

    aMaze.addRoom(r1);
    aMaze.addRoom(r2);

    r1.setSide(NORTH, factory.makewall());
    r1.setSide(EAST, aDoor);
    r1.setSide(SOUTH, factory.makewall());
    r1.setSide(WEST, factory.makewall());

    r2.setSide(NORTH, factory.makewall());
    r2.setSide(EAST, factory.makewall());
    r2.setSide(SOUTH, factory.makewall());
    r2.setSide(WEST, aDoor);
    
    return aMaze;
}
